File: src/Calendary.Model/FluxModel.cs
﻿namespace Calendary.Model;

public class FluxModel
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Version { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string ReplicateId { get; set; } = string.Empty;
    public string Status { get; set; } = "prepared"; 

    public int? CategoryId { get; set; } 
    public string? ArchiveUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }

    public bool IsPaid { get; set; } // Поле оплати

    public bool IsArchive { get; set; }

    public bool IsDeleted { get; set; }

    // Навігаційні властивості

    public WebHook? WebHook { get; set; }
    
    public User User { get; set; } = null!;

    public Category? Category { get; set; }
    public ICollection<Photo> Photos { get; set; } = new List<Photo>();
    public ICollection<Training> Trainings { get; set; } = new List<Training>();
    public ICollection<JobTask> Tasks { get; set; } = new List<JobTask>();
    public ICollection<Job> Jobs { get; set; } = new List<Job>();
}

File: src/Calendary.Ng/src/models/flux-model.ts
import { Category } from "./category";
import { Job } from "./job";
import { Photo } from "./photo";
import { Training } from "./training";
import { UserInfo } from "./user";

export class FluxModel {
    id: number = 0;
    name: string = '';
    version: string = '';
    description: string = '';
    status: string = '';
    categoryId: number = 0;
    archiveUrl?: string;
    createdAt: Date = new Date();
    completedAt?: Date;
    isPaid: boolean = false;
    trainings : Training[] = [];
    jobs: Job[] = [];
    category : Category = new Category();
  }

  export class AdminFluxModel {
    id: number = 0;
    name: string = '';
    version: string = '';
    description: string = '';
    status: string = ''; 
    categoryId: number = 0;
    archiveUrl?: string;
    createdAt: Date = new Date();
    completedAt?: Date;
    isPaid: boolean = false;
    user: UserInfo = new UserInfo();
    photos: Photo[] = [];
    trainings: Training[] = [];
    jobs: Job[] = [];
    category: Category = new Category();
  }
  

  export class CreateFluxModel {
    name: string = '';
    categoryId?: number | null = null;
  }